ArcGISRuntimeSDKQt_CppSamples/UtilityNetwork/ValidateUtilityNetworkTopology/ValidateUtilityNetworkTopology.cpp

```diff
@@ -0,0 +1,669 @@
+  m_isValidateBtnEnabled = true;
+  emit isValidateBtnEnabled();
+
+  m_isClearBtnEnabled = false;
+  emit isClearBtnEnabled();
+}
+
+void ValidateUtilityNetworkTopology::onClear()
+{
+  // Clear the selection.
+  m_deviceFeatureLayer->clearSelection();
+  m_lineFeatureLayer->clearSelection();
+
+  m_isUpdateWindowVisible = false;
+  emit isUpdateWindowVisible();
+
+  m_featureToEdit = nullptr;
+
+  updateMessage("Selection cleared.");
+
+  m_isClearBtnEnabled = false;
+  emit isClearBtnEnabled();
+}
+
+void ValidateUtilityNetworkTopology::onGetState()
+{
+  m_utilityNetwork = m_mapView ? (m_mapView->map() ? m_mapView->map()->utilityNetworks()->first() : nullptr) : nullptr;
+
+  if (m_utilityNetwork && m_utilityNetwork->definition()->capabilities()->isSupportsNetworkState())
+  {
+    m_busy = true;
+    emit isBusy();
+
+    updateMessage("Getting utility network state...");
+
+    m_utilityNetwork->stateAsync().then([this](const QFuture<Esri::ArcGISRuntime::UtilityNetworkState*>& state)
+    {
+      m_utilityNetworkstate = state.result();
+
+      // Validate if dirty areas or errors exist
+      m_isValidateBtnEnabled = m_utilityNetworkstate->hasDirtyAreas();
+      emit isValidateBtnEnabled();
+
+      // Trace if network topology is enabled
+      m_isTraceBtnEnabled = m_utilityNetworkstate->isNetworkTopologyEnabled();
+      emit isTraceBtnEnabled();
+
+      m_message = QString("Utility Network State:\n"
+          "    Has Dirty Areas: " + QString(m_utilityNetworkstate && m_utilityNetworkstate->hasDirtyAreas() ? "true" : "false") + "\n"
+          "    Has Errors: " + QString(m_utilityNetworkstate && m_utilityNetworkstate->hasErrors() ? "true" : "false") + "\n"
+          "    Is Network Topology Enabled: " + QString(m_utilityNetworkstate && m_utilityNetworkstate->isNetworkTopologyEnabled() ? "true" : "false") + "\n");
+
+      if (m_utilityNetworkstate->hasDirtyAreas() || m_utilityNetworkstate->hasErrors())
+      {
+        m_message.append("Click 'Validate' before trace or expect a trace error.");
+      }
+      else
+      {
+        m_message.append("Tap on a feature to edit or click 'Trace' to run a trace.");
+      }
+
+      updateMessage(m_message);
+    });
+  }
+
+  m_busy = false;
+  emit isBusy();
+}
+
+void ValidateUtilityNetworkTopology::onValidate()
+{  
+  m_utilityNetwork = m_mapView ? (m_mapView->map() ? m_mapView->map()->utilityNetworks()->first() : nullptr) : nullptr;
+
+  if (m_utilityNetwork)
+  {
+    // Validate using the current extent
+    const Envelope m_extent = m_mapView->currentViewpoint(ViewpointType::BoundingGeometry).targetGeometry().extent();
+
+    m_busy = true;
+    emit isBusy();
+
+    updateMessage("Validating utility network topology...");
+
+    // Get the validation result
+    UtilityNetworkValidationJob* job = m_utilityNetwork->validateNetworkTopology(m_extent);
+    job->start();
+
+    QObject::connect(job, &UtilityNetworkValidationJob::statusChanged, this, [this, job](JobStatus status)
```

Comment:
i think we don't need to specify QObject::
```suggestion
    connect(job, &UtilityNetworkValidationJob::statusChanged, this, [this, job](JobStatus status)
```